server/http/restapi/webhook.go

```diff
@@ -0,0 +1,68 @@
+package restapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+func (a *restapi) handleWebhook(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+
+	appID := vars["app_id"]
+	if appID == "" {
+		httputils.WriteBadRequestError(w, errors.New("app_id not specified"))
+		return
+	}
+
+	whName := vars["name"]
+	if whName == "" {
+		httputils.WriteBadRequestError(w, errors.New("webhook name not specified"))
+		return
+	}
+
+	queryVars := req.URL.Query()
+	if len(queryVars["secret"]) != 1 {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret was not provided"))
+		return
+	}
+
+	secret := queryVars["secret"][0]
+	if !a.validSecret(appID, secret) {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret is not valid"))
+		return
+	}
+
+	var c interface{}
+	err := json.NewDecoder(req.Body).Decode(&c)
```

Comment:
Probably an edge case, but what if the body is not JSON coded? Should we send the req.Body in a more "raw" way?

Reply:
Do you have an idea in mind? I assume there is a method available to check if `req.Body` is valid `json`?  If `req.Body` is not `json`, send as a raw `string`?

Reply:
The only thing that comes to mind is that anyone is out there using XML instead of JSON for encoding stuff. Haven't seen it on the tools we have used, but who knows...

Reply:
Why do we ever need to decode the body? Why not just pass through the bytes?

Reply:
@levb, I'm not quite sure that I understand the comment.  Right now I decode the payload into `c`, the empty interface variable. That value is the then used as part of the larger `CallResponse` object